rust/kcl-lib/src/execution/types.rs

```diff
@@ -1020,6 +1024,7 @@ impl KclValue {
             _ => self,
         };
         match ty {
+            PrimitiveType::Any => Ok(value.clone()),
```

Comment:
I think `subtype` should be altered so that `any` is a supertype of everything, and then this line is not needed

Reply:
Updating `subtype` made my tests fail in a weird way 😖 so this still needs some work.

Reply:
Because of implicit coercion of single values into arrays, we were coercing an empty `MixedArray` into `HomArray { value: vec![MixedArray { ... }] }`. Fix is here: 3a07212ae82b13aeaaae780d8b9a94a539d9a17a.